### PR TITLE
Refactor PendingBuffer to use unified ConcurrentWriteBuffer

### PR DESCRIPTION
**Pull request by lalinsky**
## Summary
- Replaced custom PendingBuffer with the same ConcurrentWriteBuffer used for main write operations
- Added max_size configuration support to prevent unbounded memory growth
- Added moveToBuffer() method for efficient data transfer between buffers
- Unified buffer architecture across the codebase

## Changes
- Remove `PendingBuffer` struct and `InsufficientBuffer` error type
- Replace `pending_buffer` field with `ConcurrentWriteBuffer` type
- Add `max_size` configuration option to `ConcurrentWriteBuffer`
- Add `moveToBuffer()`, `clear()`, and `clearInternal()` methods
- Add comprehensive tests covering moveToBuffer functionality
- Remove unused `src/buffer.zig` file

## Benefits
- **Unified Architecture**: Both write and pending buffers now use the same robust implementation
- **Thread Safety**: Pending buffer gains proper thread-safe chunked storage
- **Memory Management**: Configurable size limits prevent unbounded memory growth
- **Efficient Transfer**: moveToBuffer() method enables efficient data transfer during reconnection
- **Better Testing**: Comprehensive test coverage for new functionality

## Test Coverage
Added tests for:
- Basic moveToBuffer functionality
- Multi-chunk data transfer scenarios  
- Empty source buffer edge cases
- Buffer size limit enforcement

All existing tests continue to pass, ensuring backward compatibility.

### CODE REVIEW

File: src/queue.zig [hunk 1]
```diff
@@ -611,6 +626,53 @@ pub fn ConcurrentWriteBuffer(comptime chunk_size: usize) type {
...
+                const slice = chunk.getReadSlice();
+                if (slice.len > 0) {
+                    try dest.append(slice);
+                }
+                current = chunk.next;
+            }
+
+            // Clear the source buffer after successful move
+            self.clearInternal();
+        }
```
[the remark this answers is not in the document]
Comment (lalinsky) [reply src/queue.zig:652]: @claude see if this makes sense to implement

File: src/queue.zig [hunk 2]
```diff
@@ -611,6 +626,82 @@ pub fn ConcurrentWriteBuffer(comptime chunk_size: usize) type {
...
+            dest.queue.items_available += self.queue.items_available;
+            dest.queue.total_chunks += moved_chunk_count;
+            // Wake a waiting reader on dest (consistent with push/pushSlice)
+            dest.queue.data_cond.signal();
+
+            // Reset source queue state (we transferred ownership).
+            self.queue.head = null;
+            self.queue.tail = null;
+            self.queue.items_available = 0;
+            self.queue.total_chunks -= moved_chunk_count;
```
[the remark this answers is not in the document]
Comment (lalinsky) [reply src/queue.zig:680]: @claude evaluate this

[the remark this answers is not in the document]
Comment (lalinsky) [reply src/queue.zig:686]: @claude add the assert
